app/models/notification.rb

```diff
@@ -1,8 +1,17 @@
 class Notification < ApplicationRecord
   belongs_to :user
   belongs_to :organization
-  belongs_to :target, polymorphic: true
+  belongs_to :target, polymorphic: true, optional: true
 
+  enum subject: %i(
+    custom
+    exam_registration_enabled
+    exam_registration_approved
+    exam_registration_rejected
+    exam_results_available
+    certificate_available)
+
+  after_create :notify_via_email!
```

Comment:
I am not requesting to change this code, but i find a bit risky to send emails or have such big effects hidden on a create. The same si happening with notifications i think. We should decouple this.

Reply:
It also makes code difficult to test, and to make mistakes when doing manual tasks like migrations.

Reply:
Makes sense.